core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseCredentialsManager.scala

```diff
@@ -149,8 +151,24 @@ final class HBaseCredentialsManager private() extends Logging {
     }
   }
 
-  private def getNewToken(conf: Configuration): TokenInfo = {
-    val token = TokenUtil.obtainToken(conf)
+  private def getNewToken(conf: Configuration, currentToken: Token[_ <: TokenIdentifier]): TokenInfo = {
+    val token = {
+      if (null != currentToken) {
+        val modifiedUser = new TokenRemovingUser(currentToken)
+        var connection: Connection = null
+        try {
+          val connection = ConnectionFactory.createConnection(conf, modifiedUser)
```

Comment:
I hate variable shadowing in scala - with java we would not have had this issue.